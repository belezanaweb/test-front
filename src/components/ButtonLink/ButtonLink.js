import React from 'react';

import { Container } from './styles';

const ButtonLink = ({ children, route }) => {
  return <Container to={route}>{children}</Container>;
};

export default ButtonLink;
