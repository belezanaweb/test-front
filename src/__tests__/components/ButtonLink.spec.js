import ButtonLink from '~/components/ButtonLink';

jest.mock('react-router-dom', () => {
  return {
    Link: ({ children }) => children
  };
});

describe('Button Link Component', () => {
  it('should match a snapshot', () => {
    const wrapper = getStyledWrapperJSON(ButtonLink, {
      children: 'test'
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with route prop', () => {
    const wrapper = getStyledWrapperJSON(ButtonLink, {
      children: 'test',
      route: '/checkout'
    });

    expect(wrapper).toMatchSnapshot();
  });
});
