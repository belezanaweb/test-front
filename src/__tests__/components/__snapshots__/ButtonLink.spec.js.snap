// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link Component should match a snapshot 1`] = `"test"`;

exports[`Button Link Component should match a snapshot with route prop 1`] = `"test"`;
